cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedWorkflowProgramRunner.java

```diff
@@ -134,11 +134,11 @@ private static YarnConfiguration createConfiguration(YarnConfiguration hConf) {
    * whether the workflow contains spark.
    */
   private DriverMeta findDriverResources(Map<String, SparkSpecification> sparkSpecs,
-                                        Map<String, MapReduceSpecification> mrSpecs,
-                                        WorkflowSpecification spec) {
-    // Find the resource requirements from the workflow
+                                         Map<String, MapReduceSpecification> mrSpecs,
+                                         WorkflowSpecification spec) {
+    // Find the resource requirements from the workflow with 768MB as minimum.
     // It is the largest memory and cores from all Spark and MapReduce programs inside the workflow
-    Resources resources = new Resources();
+    Resources resources = new Resources(768);
```

Comment:
The `Resources()` has default of 512, which is easily not enough for the Workflow driver. Setting to 768 is temporary. There is work to improve resources setting for Workflow (CDAP-5066).